services/todo-legacy/controllers/todo-controller.api.ts

```diff
@@ -0,0 +1,180 @@
+export const def = {
+  basePath: "/",
+  paths: {
+    "/todo": {
+      get: {
+        "x-operation-name": "getTodo",
+        parameters: [
+          {
+            name: "id",
+            in: "query",
+            description: "The todo ID.",
+            required: true,
+            type: "string",
+            format: "JSON"
+          },
+          {
+            name: "filter",
+            in: "query",
+            description:
+              "The criteria used to narrow down the number of customers returned.",
+            required: false,
+            type: "string",
```

Comment:
AFAIK, OpenAPI spec does not support complex types for parameters `in` `query`.

See https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameter-object

> The type of the parameter. Since the parameter is not located at the request body, it is limited to simple types (that is, not an object). The value MUST be one of "string", "number", "integer", "boolean", "array" or "file". If type is "file", the consumes MUST be either "multipart/form-data", " application/x-www-form-urlencoded" or both and the parameter MUST be in "formData".